concepts/observability-strategy/configurable-monitoring/pocs/shared-vs-per-sinal-type-collector/README.md

```diff
@@ -0,0 +1,42 @@
+* Workloads may generate different amounts of signals of different types. In this case, having separate collectors allows us to scale only what is needed.
+
+### Implementation
+
+When implementing Kubernetes operators, it's common practice to have a single controller per custom resource. The Telemetry Manager is no exception. It has a `LogPipelineController` and a `TracePipelineController`. There is also a `LogParserController`, but it's not a standalone thing and is more of an extension of the `LogPipelineController`. 
+
+Each controller reconsiles it's own custom resource, translates it to a respective configuration (FluentBit or OpenTelemetry Collector), and deploys the resources. Imagine we implement a new `MetricPipelineController`, so that it shares the deployed OpenTelemetry Collector with the `TracePipelineController`. This would have some major drawbacks:
+Both controllers will have to share at least the configuration rendering code that contains both metric and tracing pipelines. If there is a bug in the configuration rendering logic, it will affect both metrics and traces
+* Both controllers will be reconciling the same resources (restarting the controller pod, etc.), possibly interfering with each other
+
+As we can see, having a shared collector also presents certain implementation challenges.
+
+### Single Endpoint
+
+From the usablity point of view it would be handy to provide a single endpoint to push telemetry data in the OTLP format. It's possible to do it using URL rewriting, so e.g. requests targeting "telemetry.kyma-system.svc.cluster.local/v1/metrics" are forwarded to the metrics gateway and "telemetry.kyma-system.svc.cluster.local/v1/traces"
+to the tracing gateway. There are a few possibilities to implement it:
+*  [Istio Virtual Service](https://istio.io/latest/docs/reference/config/networking/virtual-service/) using the default `mesh` gateway
+* An HTTP proxy with a hard-coded configuration to rewrite the request URLs
+
+The first approach is the easiest one, the only drawback is a hard dependency on Istio.
+
+### Signal Translation
+
+It is sometimes necessary to convert signals from one signal type to another. Signal translations may be direct one-to-one operations or include derivative signals such as counts, aggregations, summarizations, etc. It is possible to, let's say, convert traces to metrics in the trace gateway using the [spanmetricsprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor) and send them to the metric gateway, which in turn would send them to the backend. However, a shared collector is advantageous in this case because it can do it in-process without performing additional network calls.
```

Comment:
```suggestion
It is sometimes necessary to convert signals from one signal type to another. Signal translations may be direct one-to-one operations or include derivative signals such as counts, aggregations, summarizations, etc. It is possible to, let's say, convert traces to metrics in the trace gateway using [Span Metrics Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor) and send them to the metric gateway, which in turn would send them to the backend. However, a shared collector is advantageous in this case because it can do it in-process without performing additional network calls.
```